src/test/java/com/cloudbees/hudson/plugins/folder/FolderTest.java

```diff
@@ -137,22 +138,27 @@
         JenkinsRule.WebClient wc = r.createWebClient();
 
         // "foo" should copy "child"
-        copyFromGUI(f, wc, "foo", "xyz");
+        copyViaHttp(f, wc, "foo", "xyz");
         assertEquals("child",((Job)f.getItem("xyz")).getDescription());
         
         // "/foo" should copy "top"
-        copyFromGUI(f, wc, "/foo", "uvw");
+        copyViaHttp(f, wc, "/foo", "uvw");
         assertEquals("top",((Job)f.getItem("uvw")).getDescription());
 
     }
 
-    private void copyFromGUI(Folder f, JenkinsRule.WebClient wc, String fromName, String toName) throws Exception {
-        HtmlPage page = wc.getPage(f, "newJob");
-        ((HtmlInput)page.getElementById("name")).type(toName);
-        HtmlInput fe = (HtmlInput) page.getElementById("from");
-        fe.focus();
-        fe.type(fromName);
-        r.submit(page.getFormByName("createItem"));
+    private void copyViaHttp(Folder f, JenkinsRule.WebClient wc, String fromName, String toName) throws Exception {
+        // Taken from https://github.com/jenkinsci/jenkins/blob/80aa2c8e4093df270193402c3933f3f1f16271da/test/src/test/java/hudson/jobs/CreateItemTest.java#L68
+        r.jenkins.setCrumbIssuer(null);
+
+        URL apiURL = new URL(MessageFormat.format(
+                "{0}createItem?mode=copy&from={1}&name={2}",
+                r.jenkins.getRootUrl() + "/" + f.getUrl().toString(), fromName, toName));
```

Comment:
Why not something like "{0}/{1}create..."? Seems weird formatting a String and then format one of the parameters too.